runtime/hydradx/src/benchmarking/route_executor.rs

```diff
@@ -16,326 +16,180 @@
 	)?;
-	AssetRegistry::set_metadata(RawOrigin::Root.into(), asset_id, b"DUM".to_vec(), 18u8)?;
-
-	Ok(asset_id)
-}
-
-//NOTE: This is necessary for oracle to provide price.
-fn do_lrna_hdx_trade<T: pallet_omnipool::Config>() -> DispatchResult
-where
-	<T as pallet_omnipool::Config>::Currency: MultiCurrencyExtended<T::AccountId, Amount = i128>,
-	<T as pallet_omnipool::Config>::AssetId: From<u32>,
-{
-	let trader = create_funded_account::<T>("tmp_trader", 0, 100 * ONE, HDX.into());
-
-	fund::<T>(trader.clone(), LRNA.into(), 100 * ONE)?;
-
-	OmnipoolPallet::<T>::sell(RawOrigin::Signed(trader).into(), LRNA.into(), HDX.into(), ONE, 0)
-}
-
-//NOTE: This is necessary for oracle to provide price.
-fn do_lrna_dai_trade<T: pallet_omnipool::Config>() -> DispatchResult
-where
-	<T as pallet_omnipool::Config>::Currency: MultiCurrencyExtended<T::AccountId, Amount = i128>,
-	<T as pallet_omnipool::Config>::AssetId: From<u32>,
-{
-	let trader = create_funded_account::<T>("tmp_trader", 0, 100 * ONE, DAI.into());
 
-	fund::<T>(trader.clone(), LRNA.into(), 100 * ONE)?;
-
-	OmnipoolPallet::<T>::sell(RawOrigin::Signed(trader).into(), LRNA.into(), DAI.into(), ONE, 0)
-}
-
-fn fund<T: pallet_omnipool::Config>(
-	to: T::AccountId,
-	currency: <T as pallet_omnipool::Config>::AssetId,
-	amount: Balance,
-) -> DispatchResult {
-	CurrencyOf::<T>::deposit(currency, &to, amount)
-}
-
-use frame_support::assert_ok;
-use frame_support::traits::Hooks;
-use hydradx_traits::router::PoolType;
-use pallet_route_executor::Trade;
-use pallet_stableswap::types::AssetAmount;
-use pallet_stableswap::MAX_ASSETS_IN_POOL;
-use sp_runtime::{DispatchError, DispatchResult, FixedU128, Permill};
-use sp_std::vec;
-
-const SEED: u32 = 1;
-pub const UNITS: Balance = 100_000_000_000;
-
-fn create_funded_account<T: pallet_omnipool::Config>(
-	name: &'static str,
-	index: u32,
-	amount: Balance,
-	currency: <T as pallet_omnipool::Config>::AssetId,
-) -> T::AccountId
-where
-	<T as pallet_omnipool::Config>::AssetId: From<u32>,
-{
-	let caller: T::AccountId = account(name, index, SEED);
-
-	fund::<T>(caller.clone(), currency, amount).unwrap();
-
-	caller
-}
-
-fn set_period<T: pallet_omnipool::Config>(to: u32)
-where
-	T: pallet_ema_oracle::Config,
-	CurrencyOf<T>: MultiCurrencyExtended<T::AccountId, Amount = i128>,
-	<T as pallet_omnipool::Config>::AssetId: From<u32>,
-{
-	while System::<T>::block_number() < to.into() {
-		let b = System::<T>::block_number();
-
-		System::<T>::on_finalize(b);
-		pallet_ema_oracle::Pallet::<T>::on_finalize(b);
-
-		System::<T>::on_initialize(b + 1_u32.into());
-		pallet_ema_oracle::Pallet::<T>::on_initialize(b + 1_u32.into());
-
-		System::<T>::set_block_number(b + 1_u32.into());
-	}
+	System::set_block_number(2u32);
+	Ok(())
 }
 
 runtime_benchmarks! {
-	{ Runtime, pallet_route_executor}
+	{Runtime, pallet_route_executor}
 
+	// Calculates the weight of LBP trade. Used in the calculation to determine the weight of the overhead.
+	multi_trade_execution_sell_in_lbp {
```

Comment:
can we find a better prefix for this, instead of `multi_trade`? By reading this I expected there are multiple trade with different pools involved. Or what `multi_trade` means in this context?

Reply:
You are right that the naming is not the best. I will try to find a better name. I prefixed it with multi because these benchmarks are parameterized and can return weight for different combinations of calls (just the first one or the second one, both calls together, the first one called two times). Maybe we can rename it to something like `calculate_and_execute_buy_in_lbp(c, e)`

Reply:
that's more intention revealing, let's go for it!